crates/ruff_linter/src/rules/ruff/rules/helpers.rs

```diff
@@ -200,3 +201,25 @@ pub(super) fn is_descriptor_class(func: &Expr, semantic: &SemanticModel) -> bool
         })
     })
 }
+
+/// Returns the inferred value of the given annotation binding of a given [`Expr`].
+///
+/// If an annotation [`Expr`] is bound to a type alias, this function will return the inferred
+/// value of the type alias.
+pub(super) fn map_annotation_binding<'a>(expr: &'a Expr, semantic: &'a SemanticModel) -> &'a Expr {
+    let Some(expr_name) = expr.clone().name_expr() else {
```

Comment:
I think we can avoid the clone here
```suggestion
    let Some(expr_name) = expr.as_name_expr() else {
```